File: src/net/merayen/elastic/backend/nodes/list/signalgenerator_1/LogicNode.java
package net.merayen.elastic.backend.nodes.list.signalgenerator_1;

import net.merayen.elastic.backend.nodes.BaseLogicNode;
import net.merayen.elastic.backend.nodes.Format;
import net.merayen.elastic.system.intercom.NodeParameterMessage;

public class LogicNode extends BaseLogicNode {
	@Override
	protected void onCreate() {
		createPort(new BaseLogicNode.PortDefinition() {{
			name = "frequency";
		}});

		createPort(new BaseLogicNode.PortDefinition() {{
			name = "amplitude";
		}});

		createPort(new BaseLogicNode.PortDefinition() {{
			name = "output";
			format = Format.AUDIO;
			output = true;
		}});
	}

	@Override
	protected void onParameterChange(NodeParameterMessage message) { // Parameter change from UI
		//System.out.printf("Signalgenerator LogicNode value: %s: %s\n", message.key, message.value);
		set(message); // Acknowledge anyway
	}

	@Override
	protected void onConnect(String port) {
		System.out.println("Signalgenerator got connected");
	}

	@Override
	protected void onDisconnect(String port) {
		System.out.println("Signalgenerator got disconnected");
	}
}

File: src/main/kotlin/net/merayen/elastic/backend/logicnodes/list/poly_1/LogicNode.kt
package net.merayen.elastic.backend.logicnodes.list.poly_1

import net.merayen.elastic.backend.logicnodes.Format
import net.merayen.elastic.backend.nodes.BaseLogicNode
import net.merayen.elastic.backend.nodes.BaseNodeProperties
import net.merayen.elastic.backend.nodes.GroupLogicNode
import net.merayen.elastic.system.intercom.NodeDataMessage
import net.merayen.elastic.system.intercom.OutputFrameData

class LogicNode : BaseLogicNode(), GroupLogicNode {
	override fun onInit() {
		createInputPort("input")
		createOutputPort("output", Format.AUDIO)
	}

	override fun onParameterChange(instance: BaseNodeProperties) = updateProperties(instance) // Acknowledge anyway
	override fun onData(data: NodeDataMessage) {}
	override fun onConnect(port: String) {}
	override fun onDisconnect(port: String) {}
	override fun onRemove() {}
	override fun onFinishFrame(data: OutputFrameData?) {}
}
